### ISSUE TITLE
Sad Ghost didn't appear on levels 2-4

### ISSUE THREAD

**Issue by joshtriplett** [00-Evan/shattered-pixel-dungeon#355]
On my current game, the Sad Ghost didn't spawn. I found the Dried Rose artifact very early, but the ghost didn't appear at all. I tried quaffing a potion of mind vision, and traversed all three floors; no ghost anywhere on the map as far as mind vision could see.

I still have that game running, if there's any diagnostic information I could provide.

**Comment by 00-Evan** [comment]
Thanks for keeping the run active. If you select the run in the games in progress screen, and long-press on the hero icon on the top-left, it should print out some debug information. I can use that to try and reproduce the issue.

**Comment by joshtriplett** [comment]
On Wed, Oct 30, 2019 at 04:48:25AM -0700, Evan Debenham wrote:
> Thanks for keeping the run active. If you select the run in the games in progress screen, and long-press on the hero icon on the top-left, it should print out some debug information. I can use that to try and reproduce the issue.

Version: 0.7.5e (382)
Seed: 1033176752379
Challenge Mask: 0

**Comment by 00-Evan** [comment]
I'm not able to reproduce this. When testing with that seed the ghost consistently spawns on floor 2

**Comment by joshtriplett** [comment]
On Wed, Oct 30, 2019 at 01:23:57PM -0700, Evan Debenham wrote:
> I'm not able to reproduce this. When testing with that seed the ghost consistently spawns on floor 2 

In the still-running game, there's no ghost on floor 2. Mind vision
showed the floor completely empty.

I don't know if this makes a difference, but did you pick up the dried
rose before finding the ghost? (Only asking because this is the only
game where I've ever encountered the dried rose before the ghost.)

Also, the dried rose is in the state where the character doesn't know
what to do with it, meaning the quest hasn't been completed. Shouldn't
it be impossible for the quest to be uncompleted and the ghost to be
missing?

Is there any way I can give you the current state of the game?

**Comment by 00-Evan** [comment]
So I've dug into this a bit more, and have run into a few things which are affecting levelgen outcomes when they shouldn't, this is likely why I was getting a different result when it comes to the ghost spawning. I'll get those errors fixed (levelgen should be consistent with the same seed), but they aren't directly related to this ghost issue.

The ghost should be spawning on floor 4, and so far I can't seem to get them to fail to spawn in any cases. Just to ensure that we're getting the exact same levelgen outcomes, can you tell me this info:
- Which class were you playing?
- Do you have all guidebook pages? (including alchemy?)
- Did you identify the rose directly, using a scroll of identify, before descending to floor 4?

**Comment by joshtriplett** [comment]
On November 3, 2019 8:23:18 AM PST, Evan Debenham <notifications@github.com> wrote:
>The ghost should be spawning on floor 4, and so far I can't seem to get
>them to fail to spawn in any cases. Just to ensure that we're getting
>the exact same levelgen outcomes, can you tell me this info:
>- Which class were you playing?

Huntress.

>- Do you have all guidebook pages? (including alchemy?)

Yes.

>- Did you identify the rose directly, using a scroll of identify,
>before descending to floor 4?

No. It's still unidentified, and shows up with the purple of an item with unknown curse status.

I did directly identify several other items, though, both via identify and in one case (scrolls of upgrade) via stone of intuition.

Would screenshots of levels 2-4 and my inventory help determine if I've gotten the level generation you expect?

**Comment by 00-Evan** [comment]
Sure, floor 4 in particular is of interest as that is where the ghost should be.

**Comment by joshtriplett** [comment]
Inventory: 
![Screenshot_20191106-054423](https://user-images.githubusercontent.com/162737/68303499-d93d0b80-0058-11ea-9489-6ca70ec1ec0b.png)

Floor 2:
![Screenshot_20191106-054404](https://user-images.githubusercontent.com/162737/68303544-ebb74500-0058-11ea-8b2e-bab0ad0a5a9e.png)

Floor 3:
![Screenshot_20191106-054445](https://user-images.githubusercontent.com/162737/68303573-f671da00-0058-11ea-905e-d0da3cdc0950.png)

Floor 4:
![Screenshot_20191106-054520](https://user-images.githubusercontent.com/162737/68303590-0093d880-0059-11ea-90ca-df2346513f6b.png)

**Comment by 00-Evan** [comment]
So I'm able to correctly reproduce your run now, and I'm also using the code exactly as it was in 0.7.5e, but I'm still not able to reproduce the issue. In all cases the ghost correctly spawns on floor 4, around the middle of the bottom-right room.

I'm going to close this for now, but if you (or anyone else reading) encounters this issue again please let me know.

The inconsistent levelgen stuff is fixed indev and will be rolled out with 0.7.6.

**Comment by joshtriplett** [comment]
So, I just loaded up the game and went to floor 4, and started wandering around near the bottom right room, then around the rest of the 4th floor...and the sad ghost just showed up.

As mentioned before, the sad ghost wasn't there before (and a potion of mind vision confirmed that, showing zero other creatures on the level at all), so I'm seriously wondering how that happened. Talking to it produced the quest as normal.

So now I'm wondering down different lines: are there any circumstances under which the sad ghost or other creatures wouldn't show up on mind vision? In particular, would it matter if I quaffed mind vision on floor 2 then quickly went down to 3 then 4 while it remained active?

I'm glad that the investigation ended up turning up the issues with seeding, at least.
